### DISCUSSION TITLE
DetaBase status code 504

### DISCUSSION THREAD

**Discussion by ghost** [deta/cloud-docs#276] [Help] [answered]
I use DetaBase, but the server returns 504
Some URLs are hidden.

```python
ERROR:asyncio:Task exception was never retrieved
future: <Task finished name='Task-2571' coro=<on_message_create() done, defined at D:\Desktop\Projects\NX\main.py:4797> exception=ClientResponseError(RequestInfo(url=URL('https://database.deta.sh/v1///items/'), method='GET', headers=<CIMultiDictProxy('Host': 'database.deta.sh', 'Content-type': 'application/json', 'X-API-Key': 'c07ph2en_coXjd2ekW2LJCBrYSqTHyfKsvsrSdNNF', 'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate', 'User-Agent': 'Python/3.9 aiohttp/3.8.1')>, real_url=URL('https://database.deta.sh/v1///items/')), (), status=504, message='Gateway Time-out', headers=<CIMultiDictProxy('Date': 'Wed, 15 Jun 2022 02:29:23 GMT', 'Content-Type': 'text/html', 'Content-Length': '164', 'Connection': 'keep-alive', 'Server': 'Deta')>)>
Traceback (most recent call last):
  File "D:\Desktop\Projects\NX\main.py", line 4806, in on_message_create
    res = await translate_dict.get(str(message.guild_id))
  File "C:\Users\ikki1\AppData\Local\Programs\Python\Python39\lib\site-packages\deta\_async\client.py", line 49, in get
    raise e
  File "C:\Users\ikki1\AppData\Local\Programs\Python\Python39\lib\site-packages\deta\_async\client.py", line 43, in get
    async with self._session.get(f"{self._base_url}/items/{key}") as resp:
  File "C:\Users\ikki1\AppData\Local\Programs\Python\Python39\lib\site-packages\aiohttp\client.py", line 1138, in __aenter__
    self._resp = await self._coro
  File "C:\Users\ikki1\AppData\Local\Programs\Python\Python39\lib\site-packages\aiohttp\client.py", line 640, in _request
    resp.raise_for_status()
  File "C:\Users\ikki1\AppData\Local\Programs\Python\Python39\lib\site-packages\aiohttp\client_reqrep.py", line 1004, in raise_for_status
    raise ClientResponseError(
aiohttp.client_exceptions.ClientResponseError: 504, message='Gateway Time-out', url=URL('https://database.deta.sh/v1///items/')
```

**Comment by jnsougata** [comment]
`'X-API-Key': 'c07ph2en_coXjd2ekW2LJCBrYSqTHyfKsvsrSdNNF'` 
Not related to the issue but you've exposed your project key. Please reset it.

**Reply by ghost** [reply]
Thank you.
I thought I checked it well, but it seems that I disclosed the most important part.
Already disabled

**Comment by jnsougata** [comment]
The endpoint you are requesting ( `https://database.deta.sh/v1///items/` ) is invalid.
Correct format: `https://database.deta.sh/v1/{project_id}/{base_name}/items`
Use `PUT` for putting some data... 
Ror fetching some data by `key` the endpoint will look like `https://database.deta.sh/v1/{project_id}/{base_name}/items/{target_key}`

**Reply by ghost** [reply]
No, I thought the api key was included in it, so it was just deleted, and it was actually requested to the correct URL.
And not everything will fail.

**Reply by jnsougata** [reply]
oh! you edited the endpoint before posting it here

**Reply by jnsougata** [reply]
> No, I thought the api key was included in it, so it was just deleted, and it was actually requested to the correct URL. And not everything will fail.

Can you send some code? It seems that you have to retry on timeout

**Reply by ghost** [reply]
I use detabase's async module.

```python
@inter.event
async def on_message_create(message):
    if message.author.bot or not message.guild_id:
        return
    if message.content != "":
        if message.content.endswith("とは"):
            asyncio.create_task(message_search(message))
        if "twitter.com" in message.content:
            asyncio.create_task(message_twitter(message))
        res = await translate_dict.get(str(message.guild_id))
        if res:
            asyncio.create_task(message_translate(message, res))
```

**Reply by ghost** [reply]
![image](https://user-images.githubusercontent.com/64224091/173728352-13e6e6e7-f1d9-4b1a-9762-59260785552e.png)
You can't even browse the website with the problematic database lol

**Reply by jnsougata** [reply]
@KohnoseLami Most probaly Deta's Internal server is facing some issues which is causing down time that's why `Gateway Time-out` is happening. Not sure enough! One more thing is Deta Base's Async sdk is in alpha stage so auto retry isn't implemneted there so you have to retry manually for failure's like `504`

**Reply by ghost** [reply]
I see.
As I use DetaBase regularly, it is very troublesome for the server to stop.
But I think I can only wait.
Thank you.

**Comment by ghost** [comment, marked as answer]
This seems to be a server issue.
I didn't solve it, but I'm going to mark it as solved.

**Reply by jnsougata** [reply]
thanks
